the-collab-lab/tcl-79-smart-shopping-list
# Edit individual list items.

## Description

Users can now edit their item details directly through the "Edit Item" form. This feature allows them to modify the item name, update the next purchase date, and adjust the quantity. 

## Related Issue

closes #49 

## Acceptance Criteria

- [x] Users should be able to edit their list items details. 

## Type of Changes

Feature

## Updates

### Before

![Screen Shot 2024-10-12 at 6 11 02 PM](https://github.com/user-attachments/assets/c8fb7e0f-7b4a-4adc-8a59-62d73cf1dd05)

### After

![Screen Shot 2024-10-12 at 6 11 02 PM](https://github.com/user-attachments/assets/c8fb7e0f-7b4a-4adc-8a59-62d73cf1dd05)
![Screen Shot 2024-10-12 at 6 28 59 PM](https://github.com/user-attachments/assets/be536f9c-d97b-43a5-b76f-cb994d73785e)
![Screen Shot 2024-10-12 at 6 29 09 PM](https://github.com/user-attachments/assets/df0065ab-e8dd-4bad-ab89-668d53d920af)




## Testing Steps / QA Criteria


- Do a `git pull` and `git checkout hm-edit-listItem`.
- Open the homepage by running `npm start`.
- Navigate to the list page.
- The edit item functionality should now be in a modal triggered by the pencil icon next to the delete button.
- You can now modify the item name, update the next purchase date, and adjust the quantity.
